### PR TITLE
Fix best_fitness falsy check and NaN recovery dead-code branches in trainer

### PR DESCRIPTION
**Pull request by Zenka737**
## Summary

Replace three incorrect trainer-state semantics:

- Treat only `None` as an uninitialized `best_fitness`, preserving a real zero.
- Restore `None` when a legacy checkpoint has no `best_fitness` field, matching fresh trainer state.
- Let loss NaN/Inf, fitness NaN/Inf, or fitness collapse independently trigger the existing recovery path.

Deleted: the falsy best-fitness check, legacy `0.0` sentinel, and dead `loss_nan and (...)` conjunction; each was replaced in place with the owning state semantics.

## Validation

- Net-zero source-only diff; no tests added.
- `ruff check ultralytics/engine/trainer.py`
- `git diff --check`

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Zenka737, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This automated review has been created, and an Ultralytics engineer will assist with the next steps. Please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/engine/trainer.py [hunk 1]
```diff
@@ -816,7 +816,7 @@ def validate(self):
         if metrics is None:
             return None, None
         fitness = metrics.pop("fitness", -self.loss.detach().cpu().numpy())  # use loss as fitness measure if not found
-        if not self.best_fitness or self.best_fitness < fitness:
+        if self.best_fitness is None or self.best_fitness < fitness:
```
Comment (UltralyticsAssistant) [inline ultralytics/engine/trainer.py:819]: 💡 **MEDIUM**: This changes the meaning of the resume fallback in `_load_checkpoint_state`, which still loads a checkpoint missing `best_fitness` as `0.0`. Previously that sentinel was treated as uninitialized by the falsy check; now it is treated as a real best score. On such a resumed run, a valid negative fallback fitness (used when a validator does not provide `fitness`) will never become the best fitness, so no new `best.pt` is saved. Use `None` for an absent checkpoint field as well, or explicitly migrate the legacy fallback.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The two condition changes correctly preserve a real zero fitness and make each corruption signal independently trigger recovery. One resume-compatibility issue remains because legacy checkpoints without `best_fitness` are still initialized with the old `0.0` sentinel. Process note: the PR body also lacks the required `Deleted:` line from AGENTS.md.

💬 Posted 1 inline comment

**Comment by Zenka737** [comment]
I have read the CLA Document and I sign the CLA

**Review by glenn-jocher — APPROVED** [review]
LGTM. The final net-zero diff replaces the falsy check, legacy sentinel, and dead conjunction in place; each corruption signal now reaches the existing recovery owner independently.
